File: p22/p22.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

if __name__ == '__main__':
    f = open('names.txt')
    names = sorted(f.readline().replace('"', '').split(','))
    f.close()

    # Solution 1: using map() and lambda expression

    print sum(map(lambda name: sum([ord(c) - 64 for c in name]) \
              * (names.index(name) + 1), names))

    # Solution 2: using enumerate()

    print sum(sum([ord(c) - 64 for c in name]) * (i + 1) for (i,
              name) in enumerate(names))

File: p22/p22.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

func main() {
	f, err := os.Open("names.txt")
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	defer f.Close()

	reader := bufio.NewReader(f)
	line, _ := reader.ReadString('\n')
	names := strings.Split(strings.Replace(line, "\"", "", -1), ",")
	sort.Sort(sort.StringSlice(names))

	val := 0
	for i := 0; i < len(names); i++ {
		tmp := 0
		for j := 0; j < len(names[i]); j++ {
			tmp += int(names[i][j]) - 64
		}
		val += tmp * (i + 1)
	}

	fmt.Println(val)
}
